pkg/controller/injector/k8s_binding_injector.go

```diff
@@ -64,15 +63,10 @@ func (b *k8sBindingInjector) Inject(binding *servicecatalog.Binding, cred *broke
 		Data: make(map[string][]byte),
 	}
 
-	// For each item in the cred just serialize its value into JSON and
-	// save it in the secret
+	// For each item in the cred, just convert it to the bytes for a string
+	// representation of the value
 	for k, v := range *cred {
-		data, err := json.Marshal(v)
-		if err != nil {
-			return fmt.Errorf("Unable to marshal credential value %q: %s",
-				k, err)
-		}
-		secret.Data[k] = data
+		secret.Data[k] = []byte(fmt.Sprintf("%v", v))
```

Comment:
@duglin can you articulate what sort of types you might be concerned about? A value that's anything other than a primitive doesn't make sense here and `%v` is going to work fine for all primitive types... and if you don't buy that, then it's golang that needs more test cases; not us.